go/db_factory.go

```diff
@@ -42,9 +46,53 @@ func (f *TrinoDBFactory) CreateDB(ctx context.Context, driverName string, opts m
 		return nil, err
 	}
 
+	// Register custom HTTP client with 10-hour timeout to prevent infrastructure timeouts
+	dsn, err = f.registerCustomClientForTimeout(dsn)
+	if err != nil {
+		return nil, err
+	}
+
 	return sql.Open(driverName, dsn)
 }
 
+// registerCustomClientForTimeout creates and registers a custom HTTP client with a 10-hour timeout.
+// This prevents infrastructure/network timeouts during long-running queries
+func (f *TrinoDBFactory) registerCustomClientForTimeout(dsn string) (string, error) {
+	cfg, err := trino.ParseDSN(dsn)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse DSN: %v", err)
+	}
+
+	const httpClientTimeout = 10 * time.Hour
```

Comment:
I think we can use QueryTimeout from the DSN if set, and only use 10 hours if it's not set? (Also, trino appears to expose the timeout as a duration from its package, so can we just reference that and not hardcode it here?)